packages/backend/src/core/discovery/DiscoveryEngine.ts

```diff
@@ -1,28 +1,23 @@
-import { AddressAnalyzer } from '@l2beat/common'
-import { providers } from 'ethers'
+import { EthereumAddress } from '@l2beat/types'
 import { writeFile } from 'fs/promises'
 
 import { discover } from './discover'
 import { DiscoveryOptions } from './DiscoveryOptions'
+import { DiscoveryProvider } from './provider/DiscoveryProvider'
 import { ProjectParameters } from './types'
 
 export class DiscoveryEngine {
```

Comment:
It's getting a bit confusing to have `DiscoveryModule`, `DiscoveryEngine` and `DiscoveryProvider`. 

Since `DiscoveryEngine` became so short (that it doesn't even seem to warrant using a class), I suggest removing `DiscoveryEngine` completely and transforming it into a function somewhere.

Reply:
This comment is for the whole file, not the highlighted line :)